include/flags.h

```diff
@@ -0,0 +1,29 @@
+#ifndef FLAGS_H
+#define FLAGS_H
+
+#include <iostream>
+#include <map>
+#include <climits>
+
+class Flags {
+private:
+  int m_argc;
```

Comment:
If that's your convention make sure it's consistent. `parsed` doesn't have the prefix.